Related works and project and plan members

jdddog opened the pull request:
Migration of related works and project and plan members.

**Related works**
A lot of the values are unstructured and won't be able to be linked in the new related works system. For the first stage, it extracts DOIs from related_identifiers table with a broad regex and uses a seed table to convert other identifiers to DOIs when possible (I'm manually creating this).

I will then use the final set of plan id and DOI pairs to find records in the OpenSearch instance and create the data to be inserted in the related works tables.

**Project and plan members**
Added migrations for project and plan members and project and plan member roles.

**Misc**
- Generate external_models.yaml file, I had issues with fields using TINYINT(1), so I replaced them with BOOLEAN which appears to fix it. It also means that we don't need to define the column types in the tests for dmp tables.
- Why do we use `utf8mb4_unicode_ci`  over the default collation of `utf8mb4_0900_ai_ci`? https://dev.mysql.com/doc/refman/8.4/en/charset-unicode-sets.html

briri reviewed (approved):
This looks good @jdddog 

I tried using the external models earlier and it was giving me grief with an issue mapping DuckDB and MySQL columns. If that's no longer the case (we've made a lot of changes since then), then I think auto generating those column definitions makes sense

FYI I changed the collation to `utf8mb4_0900_ai_ci`. You may need to pull down the latest for my merge-prep branch in the backend repo and then nuke and rebuild your local DB.

There are a few files with merge conflicts now that I've merged my changes. They should be fairly straight forward to reconcile.

jdddog commented:
> This looks good @jdddog
> 
> I tried using the external models earlier and it was giving me grief with an issue mapping DuckDB and MySQL columns. If that's no longer the case (we've made a lot of changes since then), then I think auto generating those column definitions makes sense
> 
> FYI I changed the collation to `utf8mb4_0900_ai_ci`. You may need to pull down the latest for my merge-prep branch in the backend repo and then nuke and rebuild your local DB.
> 
> You will want to [merge this PR first](https://github.com/CDLUC3/dmptool-migration/pull/14) as it fixes a typo with my PR
> 
> There are a few files with merge conflicts now that I've merged my changes. They should be fairly straight forward to reconcile.

Thanks Brian, yeah the external model definition works now that we are using MySQL for the test database, with the exception of `TINYINT(1)` (it seemed to be trying to cast to `TINYINT(1)` whilst running tests and failing), so I just replaced those with `BOOLEAN`.

briri reviewed (approved):
thanks for updating the tests

briri commented:
yeah all looks good to me @jdddog I am going to merge
